platform.py

```
@@ -14,239 +14,502 @@
+            logger.error(f"Timeout in idf_tools.py after {SUBPROCESS_TIMEOUT}s")
+            return False
+        except (subprocess.SubprocessError, OSError) as e:
+            logger.error(f"Error in idf_tools.py: {e}")
+            return False
+
+    def _check_tool_version(self, tool_name: str) -> bool:
+        """Check if the installed tool version matches the required version."""
+        paths = self._get_tool_paths(tool_name)
+
+        try:
+            with open(paths['package_path'], 'r') as f:
```

Pylint (3.3.7)

[warning] Using open without explicitly specifying an encoding

(W1514)